pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -171,15 +171,16 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 		return err
 	}
 
-	// Reconcile this copy of the task run and then write back any status
+	// Reconcile this copy of the task run and then write back any status or label
 	// updates regardless of whether the reconciliation errored out.
 	err = c.reconcile(ctx, pr)
-	if equality.Semantic.DeepEqual(original.Status, pr.Status) {
-		// If we didn't change anything then don't call updateStatus.
+	if equality.Semantic.DeepEqual(original.Status, pr.Status) &&
+		reflect.DeepEqual(original.ObjectMeta.Labels, pr.ObjectMeta.Labels) {
+		// If we didn't change anything then don't call updateStatusAndLabels.
 		// This is important because the copy we loaded from the informer's
 		// cache may be stale and we don't want to overwrite a prior update
 		// to status with this stale state.
-	} else if _, err := c.updateStatus(pr); err != nil {
+	} else if _, err := c.updateStatusAndLabels(pr); err != nil {
```

Comment:
I'm not sure about this and the equivalent in `taskrun.go`. Before #367, both of these places were using `.Update` in their corresponding `updateStatus` functions, and I don't know if switching from `.UpdateStatus` back to `.Update` will break anything that that PR was doing. Would it be better to update the PipelineRun's labels immediately in the reconciler? Any ideas of how to avoid recomputing labels? For consistency it would be nice to actually set the labels on the `PipelineRun`/`TaskRun` but it feels a little awkward to update parts of those objects that are not the status.

Reply:
I think this is the change causing the tests to fail. Using `Update` does not update the `Status` even if the status field has changed, you have to use `UpdateStatus` to change the status. It looks like this is explained in [the docs for CRDs](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#status-subresource): 

> [If the status subresource is enabled] PUT/POST/PATCH requests to the custom resource ignore changes to the status stanza

I'll update the code to use `Update` to modify labels and leave `Status` handling as it was before. 

Also I noticed that #516 switched from `UpdateStatus` to `Update` in `cancel.go`, which might run into the same problem (or maybe it was a fix because the status in that code is not the actual status subresource since it is on the TaskRun's spec rather than TaskRun itself?).

Reply:
#516 is only changing this for the cancel case which is different because the status in that code is not the actual status subresource like you said